homeassistant/components/sun.py

```diff
@@ -30,6 +30,10 @@
 
 STATE_ATTR_AZIMUTH = 'azimuth'
 STATE_ATTR_ELEVATION = 'elevation'
+STATE_ATTR_NEXT_DAWN = 'next_dawn'
+STATE_ATTR_NEXT_DUSK = 'next_dusk'
+STATE_ATTR_NEXT_MIDNIGHT = 'next_midnight'
```

Comment:
My use case: transition light.color_temp mostly between dawn and sunrise (and sunset and dusk) and also a bit between sunrise and noon (and noon and sunset).
The Flux component just isn't flexible enough for me 👼 

Talking about 'adding things': I find it rather confusing what's considered "addible" and what not.
It seems to me that components like [Flux](https://home-assistant.io/components/switch.flux/) and [Presence Based Lights](https://home-assistant.io/components/device_sun_light_trigger/) could be easily made as (template based) automations (or custom_components outside the core). While exposing attributes like these actually offer users more flexibility.
It's not like I want to question the decision making, but some kind of guide/docs about what is considered addible/mergable and what's not would save me time and effort on making PR's, tests and documentation and might also reduce the number of reviews you'll have to do 😉